File: typescript/src/note.ts
import { Kylrix } from './index';

/**
 * Kylrix.Note: The Intelligence Layer Module.
 * Domain: note.kylrix.space
 */
export class KylrixNote {
  constructor(private sdk: Kylrix) {}

  /**
   * Saves a note revision with optional metadata.
   */
  async saveRevision(databaseId: string, tableId: string, data: {
    noteId: string;
    userId: string;
    content: string;
    title?: string;
    diff?: string;
    cause?: 'manual' | 'ai' | 'collab';
  }) {
    return await this.sdk.createRow(databaseId, tableId, {
      ...data,
      createdAt: new Date().toISOString(),
      cause: data.cause || 'manual',
    });
  }

  /**
   * Adds a tag to a note.
   */
  async addTag(databaseId: string, tableId: string, noteId: string, tagName: string) {
    const note = await this.sdk.getRow<any>(databaseId, tableId, noteId);
    const tags = new Set(note.tags || []);
    tags.add(tagName);
    
    return await this.sdk.updateRow(databaseId, tableId, noteId, {
      tags: Array.from(tags),
    });
  }

  /**
   * Lists notes for a specific user.
   */
  async listNotes(databaseId: string, tableId: string, userId: string) {
    return await this.sdk.listRows<any>(databaseId, tableId, [
      `equal("userId", "${userId}")`
    ]);
  }
}

File: dart/lib/src/note.dart
import 'client.dart';

/// Kylrix.Note: The Intelligence Layer Module.
/// Domain: note.kylrix.space
class KylrixNote {
  final Kylrix _sdk;

  KylrixNote(this._sdk);

  /// Saves a note revision.
  Future<dynamic> saveRevision({
    required String databaseId,
    required String tableId,
    required String noteId,
    required String userId,
    required String content,
    String? title,
    String? diff,
    String cause = 'manual',
  }) async {
    return await _sdk.createRow(
      databaseId: databaseId,
      tableId: tableId,
      data: {
        'noteId': noteId,
        'userId': userId,
        'content': content,
        'title': title,
        'diff': diff,
        'cause': cause,
        'createdAt': DateTime.now().toIso8601String(),
      },
    );
  }

  /// Adds a tag to a note.
  Future<dynamic> addTag({
    required String databaseId,
    required String tableId,
    required String noteId,
    required String tagName,
  }) async {
    final note = await _sdk.getRow(
      databaseId: databaseId,
      tableId: tableId,
      rowId: noteId,
    );

    final List<String> tags = List<String>.from(note.data['tags'] ?? []);
    if (!tags.contains(tagName)) {
      tags.add(tagName);
    }

    return await _sdk.updateRow(
      databaseId: databaseId,
      tableId: tableId,
      rowId: noteId,
      data: {'tags': tags},
    );
  }
}
